metricbeat/mb/testing/data/data_test.go

```diff
@@ -51,12 +51,14 @@ const (
 var (
 	// Use `go test -generate` to update files.
 	generateFlag = flag.Bool("generate", false, "Write golden files")
+	moduleFlag   = flag.String("module", "", "Write golden files")
 )
 
 type Config struct {
-	Type   string
-	URL    string
-	Suffix string
+	Type         string
+	URL          string
+	Suffix       string
+	ModuleConfig map[string]interface{} `yaml:"module_config"`
```

Comment:
I wonder if we should call it just `config`?

Reply:
The idea was that a reader can clearly understand that anything in this field is _module related configuration_ and not _test related configuration_.

What about calling it just `module` then? If the reason is to short it, the file is already called `config.yml` so the fact that some configuration is going to be find inside is assumed. 
* `config` only could lead to confusion as soon as we need any other config, for example metricset config (so we don't end up with `config`, which refers to module config and `metricsetConfig` for something specific of the metricset). 
* An alternative is to call it `config.Module` so later we can also have `config.Metricset` but I think that as the file is called `config.yml`, prefixing a field with `config.` is redundant as well as placing it in the suffix like now.

At the same time, if we maintain the variables names like `config` in `data_test.go` code, calls to this module configuration will look like `config.Module` instead of `config.Config`.